app/src/main/kotlin/com/intive/picover/parties/repository/PartiesRepository.kt

```diff
@@ -20,6 +21,13 @@ class PartiesRepository @Inject constructor(
 			}
 		}
 
+	suspend fun addParty(partyRemote: PartyRemote) =
+		runCatching {
+			partiesReference.push().let {
+				it.setValue(partyRemote.copy(id = it.key ?: "")).await()
```

Comment:
in what case the key will be null? I'm worried empty string could break something elsewhere

Reply:
According to the documentation "
The last token in the location pointed to by this reference or null if this reference points to the database root". So very unlikely, but I get the point. 
Code updated.